code/API_definitions/device-status.yaml

```diff
@@ -1,7 +1,72 @@
+  description: |
+    This API provides the customer with the ability to query the roaming status of a device, and to subscribe to receive notifications when the device roaming status changes.
+
+    # Introduction
+
+    The API consumer is able to verify whether a certain user device is in a roaming situation (or not). This capability is provided in two ways:
+      - via a direct request, with the current roaming status given in the response.
+      - via a subscription request - in this case the roaming situation is not in the response but a notification is sent back to the subscriber when the device roaming status changes
+
+    The verification of the roaming situation depends on the network's ability. In addition to the roaming status, visited country information is included in the response when the device is roaming.
+
+    Device roaming status verification could be useful in scenarios such as (not exhaustive):
+      - For regulatory reasons, where a customer may need to be within a certain jurisdiction, or outwith others, in order for transactions to be authorized
+      - For security / fraud / fair play reasons, to establish that a customer is located where they claim to be
+      - For service delivery reasons, to ensure that the customer has access to particular service, and will not incur roaming charges in accessing them
+
+
+    # Relevant terms and definitions
+
+    * **Device**: A device (i.e. user equipment) refers to any physical entity that can connect to a mobile network and participate in network communication.
+      At least one identifier for the device must be provided from the following four options:
+      - IPv4 address
+      - IPv6 address
+      - Phone number (MSISDN)
+      - Network Access Identifier assigned by the mobile network operator for the device
+
+    * **Roaming** : Roaming status:
+      - `true` if device is connected to a network other than its home network
+      - `false` if connected to its home network
+
+    * **Country** : Visited country code and name:
+      - Mobile country code (MCC), as defined by ITU-T E.212 (e.g. `262`).
+      - ISO country code, as defined by ISO 3166 ALPHA-2 (e.g. `DE`). A list may be provided if multiple countries share the same MCC.
+
+
+    # API Functionality    
+
+    The API exposes following capabilities:
+
+    ## Device roaming status
+
+    The endpoint `POST /roaming` allows the APi consumer to synchronously (i.e. immediately) get roaming status and country information (if the device is roaming).
+
+    ## Device status event subscription
+
+    This endpoint allows the API consumer to create and delete event subscriptions for specific roaming device status events. The CAMARA subscription model is detailed in the [CAMARA API Design Guidelines](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md).
```

Comment:
I updated the links to be hash-links to the most appropriate numbered section in the design guidelines. Unfortunately the sections on instance-based and resource-based subscriptions are not numbered, so cannot be directly hash-linked. This could be changed.

My preference is not to copy over information from the design guidelines, because then we have a maintenance issue whenever the design guidelines are updated. Of course, by hash-linking there is still an issue if the section names or numbers change, but probably more manageable.

Anyway, that's my proposal, but happy to discuss.